include/openPMD/RecordComponent.hpp

```diff
@@ -275,6 +324,16 @@ RecordComponent::loadChunk(std::shared_ptr< T > data, Offset o, Extent e, double
         dRead.dtype = getDatatype();
         dRead.data = std::static_pointer_cast< void >(data);
         m_chunks->push(IOTask(this, dRead));
+
+        if( this->IOHandler->flushType == FlushType::DIRECT )
+        {
+            while( !m_chunks->empty() )
```

Comment:
The loop is probably redundant. The only scenario in which there would be more than one chunks in the `m_chunks` queue is when one chunk (or more) is scheduled in `DEFER`, the Series is switched to `DIRECT` and another chunk is scheduled.

Reply:
Yes, that could happen imho.